### PR TITLE
Improve UI for ontology search and access token generation

### PR DESCRIPTION
**Pull request by wow-such-code**
* improves naming and description for ontology search
* actually counts available ontologies now
* indicates successful copying of the access token, as well as the ontology curie
* resets indicators after 1 second using a background thread

### CODE REVIEW

File: project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java [hunk 1]
```diff
@@ -27,4 +27,6 @@ Page<OntologyClass> findByLabelFulltextMatching(
 
   List<OntologyClass> findOntologyClassEntitiesByCurie(String name);
 
+  @Query(value = "SELECT DISTINCT ontology from ontology_classes;", nativeQuery = true)
+  List<String> findUniqueOntologies();
```
Comment (KochTobi) [inline project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java:31]: A JPA interface should always return the same type. In your case I would expect a list of OntologyClasses returned.

File: project-management/src/main/java/life/qbic/projectmanagement/application/ontology/OntologyLookupService.java [hunk 2]
```diff
@@ -51,4 +51,8 @@ public Optional<OntologyClass> findByCURI(String curie) {
     return ontologyTermLookup.query(new OntologyCurie(curie)).stream().findAny();
   }
 
+  public List<String> findNumberOfOntologies() {
```
Comment (KochTobi) [inline project-management/src/main/java/life/qbic/projectmanagement/application/ontology/OntologyLookupService.java:54]: The method is called findNumber... so why not return a long or int instead?

File: user-interface/src/main/java/life/qbic/datamanager/views/account/PersonalAccessTokenComponent.java [hunk 3]
```diff
@@ -120,11 +120,14 @@ private void showCreatedPersonalAccessToken(String rawTokenText) {
...
     copyIcon.addClassName(IconSize.SMALL);
     copyIcon.addClassName("clickable");
-    copyIcon.addClickListener(
-        event -> UI.getCurrent().getPage().executeJs("window.copyToClipboard($0)", rawTokenText));
+
     Span rawToken = new Span(rawTokenText);
     Span personalAccessTokenWithIcon = new Span(rawToken, copyIcon);
     personalAccessTokenWithIcon.addClassName("token-text");
+
+    copyIcon.addClickListener(
```
Comment (KochTobi) [inline user-interface/src/main/java/life/qbic/datamanager/views/account/PersonalAccessTokenComponent.java:128]: NIT: why did you move this down? I think it makes more sense to keep the listener close to the icon for which it is added.

File: project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepository.java [hunk 4]
```diff
@@ -93,4 +93,8 @@ private static Sort sortByOrders(List<SortOrder> sortOrders) {
             : Order.asc(sortOrder.propertyName()))
         .toList());
   }
+
+  public List<String> findUniqueOntologies() {
```
Comment (KochTobi) [inline project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepository.java:97]: doesn't find unique ontologies but only unique ontology abbreviations

File: user-interface/src/main/java/life/qbic/datamanager/views/account/PersonalAccessTokenComponent.java [hunk 5]
```diff
@@ -137,6 +140,34 @@ private void showCreatedPersonalAccessToken(String rawTokenText) {
     updateUI();
   }
 
+private void handleCopyClicked(String rawTokenText) {
```
Comment (KochTobi) [inline user-interface/src/main/java/life/qbic/datamanager/views/account/PersonalAccessTokenComponent.java:143]: this effectively prevents users from copying the token multiple times. This is bad IMHO because if I accidentally overwrite it in my clipboard I lost the ability to copy it easily. How about leaving the copy icon in place and only changing text and background color?

File: user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java [hunk 6]
```diff
@@ -159,13 +160,36 @@ private Span createHeader(String label, String curieText) {
...
-      Span header = new Span(title, curie, initCopyIcon(curieText));
+          event -> handleCopyClicked(header, curieText));
+      copyIcon.addClickListener(
+          event -> handleCopyClicked(header, curieText));
+
       header.addClassName("header");
       return header;
     }
 
+    private void handleCopyClicked(Span header, String curieText) {
```
Comment (KochTobi) [inline user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java:176]: Once I copied an ontology, I cannot do so again. This is bad. I want to copy it into multiple fields in my measurement sheet. If I copy a second term, the previous is still highlighted (do we want this?)

**Review by KochTobi — CHANGES_REQUESTED** [review]

File: project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java (hunk 1 above)
Comment (wow-such-code) [reply project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java:31]: Unfortunately that won't be possible using the DISTINCT keyword. Alternatives would be not to use the feature or create another interface?

Comment (KochTobi) [reply project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/OntologyTermRepositoryJpaInterface.java:31]: you are right.
